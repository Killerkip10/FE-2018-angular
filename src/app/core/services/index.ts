export * from './auth.service';
export * from './profile.service';
export * from './token.service';
export * from './search-user.service';
