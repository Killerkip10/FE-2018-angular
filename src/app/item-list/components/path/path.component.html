<mat-form-field>
  <input matInput [ngModel]="path" placeholder="Path" disabled>
</mat-form-field>
