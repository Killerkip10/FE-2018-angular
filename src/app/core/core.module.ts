import {NgModule} from '@angular/core';
import {BrowserModule} from '@angular/platform-browser';
import {TranslateModule} from '@ngx-translate/core';
import {HttpClientModule} from '@angular/common/http';
import {RouterModule} from '@angular/router';

import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
import {AuthGuard} from './guards';
import {AuthService} from './services';
import {
  MatButtonModule,
  MatToolbarModule,
  MatSelectModule,
  MatIconModule
} from '@angular/material';
import {
  HeaderComponent,
  FooterComponent
} from './components';

@NgModule({
  imports: [
    BrowserModule,
    HttpClientModule,
    TranslateModule,
    BrowserAnimationsModule,

    RouterModule,

    MatIconModule,
    MatButtonModule,
    MatSelectModule,
    MatToolbarModule
  ],
  declarations: [
    HeaderComponent,
    FooterComponent
  ],
  providers: [
    AuthService,

    AuthGuard
  ],
  exports: [
    HeaderComponent,
    FooterComponent
  ]
})
export class CoreModule {}
