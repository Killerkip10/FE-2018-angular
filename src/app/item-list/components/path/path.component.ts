import {ChangeDetectionStrategy, Component, Input} from '@angular/core';

@Component({
  selector: 'app-path',
  templateUrl: './path.component.html',
  styleUrls: ['./path.component.scss'],
  // changeDetection: ChangeDetectionStrategy.OnPush
})
export class PathComponent {
  @Input() public path: string;

  constructor() { }
}
