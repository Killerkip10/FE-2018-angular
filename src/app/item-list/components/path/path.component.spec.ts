import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import {ChangeDetectionStrategy, NO_ERRORS_SCHEMA} from '@angular/core';

import { PathComponent } from './path.component';

describe('PathComponent', () => {
  let component: PathComponent;
  let fixture: ComponentFixture<PathComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        PathComponent
      ],
      schemas: [
        NO_ERRORS_SCHEMA
      ]
    })
      .overrideComponent(PathComponent, {set: {changeDetection: ChangeDetectionStrategy.Default}})
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(PathComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.path = 'item1 // item 2 // item 3';
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
