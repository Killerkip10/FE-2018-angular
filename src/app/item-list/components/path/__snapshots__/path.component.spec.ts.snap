// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PathComponent snapshot 1`] = `
<app-path>
  <mat-form-field>
    <input
      disabled=""
      matinput=""
      placeholder="Path"
    />
  </mat-form-field>
</app-path>
`;

exports[`PathComponent snapshot 2`] = `
<app-path
  path={[Function String]}
>
  <mat-form-field>
    <input
      disabled=""
      matinput=""
      placeholder="Path"
    />
  </mat-form-field>
</app-path>
`;
